src/client/ua_client.c

```diff
@@ -20,6 +21,15 @@
 static void UA_Client_init(UA_Client* client, UA_ClientConfig config) {
     memset(client, 0, sizeof(UA_Client));
     client->channel.connection = &client->connection;
+    client->channel.endpoint = (UA_Endpoint*)UA_malloc(sizeof(UA_Endpoint));
```

Comment:
Okay got it. This will be reviewed anyway, when the client gets an extended _connect public API where the endpoint can be selected.